html/views/overlays/interactive/index.js

```diff
@@ -1,38 +1,101 @@
-		       "ScoreBoard.Settings.Setting(ScoreBoard.Intermission.Unofficial)",
-		       "ScoreBoard.Settings.Setting(ScoreBoard.Intermission.Official)",
-		       "ScoreBoard.Settings.Setting(ScoreBoard.Intermission.Intermission)" ], function(k,v) { } );
-
-	WS.Register( [  'ScoreBoard.Clock(Timeout).Running', 
-		        'ScoreBoard.TimeoutOwner',
+			"ScoreBoard.Rulesets.CurrentRule(Period.Number)",
+			"ScoreBoard.Settings.Setting(ScoreBoard.Intermission.PreGame)",
+			"ScoreBoard.Settings.Setting(ScoreBoard.Intermission.Unofficial)",
+			"ScoreBoard.Settings.Setting(ScoreBoard.Intermission.Official)",
+			"ScoreBoard.Settings.Setting(ScoreBoard.Intermission.Intermission)" ]);
+
+	WS.Register( [	'ScoreBoard.Clock(Timeout).Running',
+			'ScoreBoard.TimeoutOwner',
 			'ScoreBoard.OfficialReview',
-			'ScoreBoard.Team(1).Timeouts',
-			'ScoreBoard.Team(2).Timeouts',
-			'ScoreBoard.Team(1).OfficialReviews',
-			'ScoreBoard.Team(2).OfficialReviews',
-			'ScoreBoard.Team(1).RetainedOfficialReview',
-			'ScoreBoard.Team(2).RetainedOfficialReview' ], function(k,v) { smallDescriptionUpdate(k,v); } );
+			'ScoreBoard.Team(*).Timeouts',
+			'ScoreBoard.Team(*).OfficialReviews',
+			'ScoreBoard.Team(*).RetainedOfficialReview' ], function(k,v) { smallDescriptionUpdate(k,v); } );
+
+	WS.Register( ['ScoreBoard.Period(*).Jam(*).TeamJam(*).JamScore',
+			'ScoreBoard.Period(*).Jam(*).TeamJam(*).Lead',
+			'ScoreBoard.Period(*).Jam(*).TeamJam(*).Lost'], function(k,v) { jamData(k,v); } );
+
+	WS.Register( [
+			'ScoreBoard.Team(*).Skater(*).Name',
+			'ScoreBoard.Team(*).Skater(*).Number',
+			'ScoreBoard.Team(*).Skater(*).Flags' ], function(k,v) {
+				var me = '.RosterTeam' + k.Team + ' .Team' + k.Team + ' .Skater[data-skaterId=' + k.Skater + ']';
+				var mb = '.PenaltyTeam' + k.Team + ' .Team' + k.Team + ' .Skater[data-skaterId=' + k.Skater + ']';
+				if (v == null) {
+					$(me).remove();
+					$(mb).remove();
+					return;
+				}
+				ensureSkaterExists(k.Skater, k.Team);
+
+				if(k.field == 'Number')	{ updateSkaterNumber(me,mb,v); }
+				if(k.field == 'Flags') {
+					$('.'+k.field, me).attr('data-flag', v);
+					$(mb).attr('data-flag', v);
+				} else if (v == "") {
+					c = $('.'+k.field, me).html('&nbsp;');
+					c = $('.'+k.field, mb).html('&nbsp;');
+				} else {
+					c = $('.'+k.field, me).text(v);
+					c = $('.'+k.field, mb).text(v);
+				}
+			});
+
+	WS.Register( 'ScoreBoard.Team(*).Skater(*).Penalty(*).Code', function(k,v) {
+		if (k.Penalty == 0) {
+			// Foulout/Expulsion.
+			return;
+		}
+		var sel = '.PenaltyTeam' + k.Team + ' .Team' + k.Team + ' .Skater[data-skaterId=' + k.Skater + ']';
+		if (v == null) {
+			$('.Number-'+k.Penalty, sel).remove();
+			$(sel).attr('data-count', $('.Penalty', sel).length);
+			return;
+		}
+		ensureSkaterExists(k.Skater, k.Team);
+		createPenalty(sel, k.Penalty, v);
+	} );
+
+	WS.Register( 'ScoreBoard.Team(*).Color(*)', function(k,v) {
```

Comment:
The function only acts on `overlay_bg` and `overlay_fg` - wouldn't it make more sense to only register those keys.

Reply:
Done